journal/202310.md

```diff
@@ -0,0 +1,766 @@
+# {DRAFT} Decred Journal – October 2023
+
+![](../img/ { _upload 202310.01 title image_ })
+
+_Image: { "Title by @author. Description..." OR just "Description..." }_
+
+{ _Month's highlights, no links here, 5-7 bullets max._
+
+- Summarize Cake Wallet integration activity for Oct - prop approved, repos setup, work started
+
+- BLAKE3 GPU pool mining enabled with gominer and dcrpool upgrades.
+
+- New ticket price ATH
+
+- 
+
+- 
+
+}
+
+Contents: { _Sync ToC at the very end, when all text is finished_ }
+
+- [Development](#development)
+- [People](#people)
+- [Governance](#governance)
+- [Network](#network)
+- [Ecosystem](#ecosystem)
+- [Outreach](#outreach)
+- [Events](#events)
+- [Media](#media)
+- [Markets](#markets)
+- [Relevant External](#relevant-external)
+
+
+<a id="{}" />
+
+## Core Software v1.8.1 Release
+
+{ 
+
+- https://github.com/decred/decred-binaries/releases/tag/v1.8.1
+- https://github.com/decred/decred-release/releases/tag/v1.8.1
+
+}
+
+
+<a id="{some-announcement}"></a>
+
+## { Some Announcement }
+
+{ _Add individual sections for high-impact announcements, such as consensus votes, software releases, milestones, major bugs, etc._ }
+
+
+<a id="development"></a>
+
+## Development
+
+The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.
+
+
+### dcrd
+
+_[dcrd](https://github.com/decred/dcrd) is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+User-facing changes:
+
+- Added support for [concurrent `getdata` requests](https://github.com/decred/dcrd/pull/3203). Currently a node can process only one `getdata` request at a time, a behavior inherited from very early network code that is no longer relevant. While the data for one `getdata` request is being served, other messages that would otherwise be fast to process, such as pings, can potentionally be delayed for a long time. Moreover, it is theoretically possible for a pair of peers to get stuck waiting for each other. In practice this situation basically never happens but this may change in the future when new messages are introduced such as those for peer-to-peer mixing. To improve overall throughput and address the aforementioned concerns, `getdata` requests have been made asynchronous so other inbound messages can be processed concurrently. New limits have been introduced as well in order to block malicious behavior. As a bonus, concurrent `getdata` requests speed up [initial sync](https://matrix.to/#/!zefvTnlxYHPKvJMThI:decred.org/$mxemr1HGLNlX4eXYPuG9rTVUb15Jg7lWudGK3TQJEG0). This change to the wire protocol is planned for inclusion in the v1.9.0 release.
+  - { _does this one change cut IBD time from 1h21m down to 47m?!_ }
```

Comment:
Correct.

Reply:
To expand a bit, the reason is that it significantly reduces the latency.

Prior to this change, due to the fact the request queue would be blocked until the entire batch of requested blocks was received, several other messages would accumulate in the queue that would have to be serviced before the request for the next batch of blocks was sent.  Since there is network latency involved with every such request, it wasn't uncommon for it to add up to ~5 seconds of what amounts to "idle" time between every batch of blocks.

Given there are 800k+ blocks and they're typically retrieved in batches of 2000, a full initial sync would incur 400+ such delays.  In other words, about 2000 seconds or ~33 minutes.

This change effectively removes all of that latency by allowing the queue to serviced concurrently while the node is otherwise busy reading/writing blocks.

Reply:
Massive time savings. Thanks for explaining!